packages/composables/src/hooks/useCart/index.ts

```diff
@@ -2,23 +2,32 @@ import { ref, Ref, computed } from "@vue/composition-api";
 import {
   getCart,
   addProductToCart,
+  addPromotionCode,
   removeCartItem,
   changeCartItemQuantity,
 } from "@shopware-pwa/shopware-6-client";
 import { ClientApiError } from "@shopware-pwa/commons/interfaces/errors/ApiError";
 import { Cart } from "@shopware-pwa/commons/interfaces/models/checkout/cart/Cart";
 import { Product } from "@shopware-pwa/commons/interfaces/models/content/product/Product";
 import { LineItem } from "@shopware-pwa/commons/interfaces/models/checkout/cart/line-item/LineItem";
-import { getApplicationContext } from "@shopware-pwa/composables";
+import {
+  getApplicationContext,
+  useNotifications,
+} from "@shopware-pwa/composables";
 import { ApplicationVueContext } from "../../appContext";
 
+const TYPE_PROMOTION = "promotion";
+const TYPE_PRODUCT = "product";
+
 /**
  * interface for {@link useCart} composable
  *
  * @beta
  */
 export interface IUseCart {
   addProduct: ({ id, quantity }: { id: string; quantity?: number }) => void;
+  addPromotionCode: (promoCode: string) => void;
+  appliedPromotionCodes: Readonly<Ref<Readonly<LineItem[]>>>;
```

Comment:
we can use now `ComputedRef` instead of this typing :)
```suggestion
  addPromotionCode: (promoCode: string) => Promise<void>;
  appliedPromotionCodes: ComputedRef<LineItem[]>;
```